shard.go

```diff
@@ -79,7 +126,14 @@ type Shard struct {
 }
 
 // newShard returns a new initialized Shard instance.
-func newShard() *Shard { return &Shard{} }
+func newShard(id uint64) *Shard {
+	s := &Shard{}
+	s.ID = id
+	s.stats = NewStats(fmt.Sprintf("shard %d", id))
+	s.stats.Inc("initialize")
```

Comment:
This stat is a complete hack for debugging right now.  If this does indeed fix the problem, we should remove it.  It did already help as it showed me that a shard had been created, opened, and closed, and they tried to read from a snapshot.  So, if this fixes the bug, then it has outlived it's purpose (which it served well for).